apps/test-suite/tests/FileSystemNext.ts

```diff
@@ -489,6 +489,14 @@ export async function test({ describe, expect, it, ...t }) {
           }
           expect(error.message.includes('Destination already exists')).toBe(true);
         });
+
+        it('downloads file when headers are set', async () => {
+          const url = 'https://picsum.photos/id/237/200/300';
+          const file = new File(testDirectory, 'image.jpeg');
+          const output = await File.downloadFileAsync(url, file, { headers: { Token: '1234' } });
```

Comment:
this test doesn't test too much, but don't really have an idea on how to also test headers while downloading without hosting our own server.